sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -770,40 +760,35 @@ class Analyzer(
 
   object ResolveInsertInto extends Rule[LogicalPlan] {
     override def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
-      case i @ InsertIntoStatement(
```

Comment:
Since we are cleaning up the code here, how about we make the responsibility of each rule more clear? What I have in my mind is:
1. `ResolveInsertInto` should only resolve the `InsertIntoStatement`, instead of the relation itself. Ideally it should only match `InsertIntoStatement(DataSourceV2Relation, ...)`, and let other rules to resolve relations.
2. `ResolveTables` should resolve `UnresolvedRelation` inside `InsertIntoStatement`.